chia/full_node/mempool_manager.py

```diff
@@ -406,11 +406,7 @@ async def add_spendbundle(
 
             if error:
                 if error is Err.ASSERT_HEIGHT_ABSOLUTE_FAILED or error is Err.ASSERT_HEIGHT_RELATIVE_FAILED:
-                    potential = MempoolItem(
-                        new_spend, uint64(fees), npc_result, cost, spend_name, additions, removals, program
-                    )
-                    self.add_to_potential_tx_set(potential)
```

Comment:
a `SpendBundle` that fails because of a time-lock or height-lock isn't inherently invalid, it just can't be included in a block *yet*.

we add these toe the potential set to allow the farmer to include them in the future, when their time-lock has expired.

Reply:
that said, it would probably make sense to have an upper limit on how long is left on a time-lock. If the spend bundle won't be valid for years, it probably doesn't make sense to include it for instance.

Reply:
Thanks for your feedback. I agree that my proposal is not the best solution. It would be best to ensure mempool makes sure all SpendBundles are sane and won't leave mempool in an inconsistent state (which is what has happened in #7720).

The second best solution would be to make mempool self-healing. If an inconsistent state has been detected, try to correct it or reinitialize mempool, so the node isn't left in a broken state (causing mass-disconnects and sync issues).

As I haven't yet fully understood the root cause, I suggested the suboptimal workaround in this PR. At least this ensures that this current cause of mempool corruption won't happen again.

I have a hunch that leaving this as is for now (pre-PR) may enable potentially malicious nodes to intentionally corrupt other nodes' mempools. This in turn will cause those corrupted nodes to ban their peer nodes which can cause sync issues for the whole blockchain. That's the reason I went for this quick fix, but I'm happy if wiser people come up with a better solution.

Reply:
Thanks for your work on this skweee! We will probably not take this PR, but are working on a PR that will address this thanks to your research.